esphome/components/micronova/text_sensor/micronova_text_sensor.cpp

```diff
@@ -8,21 +8,15 @@ void MicroNovaTextSensor::process_value_from_stove(int value_from_stove) {
     return;
   }
 
-  switch (this->get_function()) {
-    case MicroNovaFunctions::STOVE_FUNCTION_STOVE_STATE:
-      this->micronova_->set_current_stove_state(value_from_stove);
-      this->publish_state(STOVE_STATES[value_from_stove]);
-      // set the stove switch to on for any value but 0
-      if (value_from_stove != 0 && this->micronova_->get_stove_switch() != nullptr &&
-          !this->micronova_->get_stove_switch()->get_stove_state()) {
-        this->micronova_->get_stove_switch()->set_stove_state(true);
-      } else if (value_from_stove == 0 && this->micronova_->get_stove_switch() != nullptr &&
-                 this->micronova_->get_stove_switch()->get_stove_state()) {
-        this->micronova_->get_stove_switch()->set_stove_state(false);
-      }
-      break;
-    default:
-      break;
+  this->micronova_->set_current_stove_state(value_from_stove);
+  this->publish_state(STOVE_STATES[value_from_stove]);
+  // set the stove switch to on for any value but 0
+  if (value_from_stove != 0 && this->micronova_->get_stove_switch() != nullptr &&
+      !this->micronova_->get_stove_switch()->get_stove_state()) {
+    this->micronova_->get_stove_switch()->set_stove_state(true);
+  } else if (value_from_stove == 0 && this->micronova_->get_stove_switch() != nullptr &&
+             this->micronova_->get_stove_switch()->get_stove_state()) {
+    this->micronova_->get_stove_switch()->set_stove_state(false);
   }
```

Comment:
```suggestion
  auto stove_switch = this->micronova_->get_stove_switch();
  if (stove_switch != nullptr) {
    if (value_from_stove != 0 && !stove_switch->get_stove_state()) {
      stove_switch->set_stove_state(true);
    } else if (value_from_stove == 0 && stove_switch->get_stove_state()) {
      stove_switch->set_stove_state(false);
    }
  }
```
nit: might be easier to read

Reply:
merged already #12355 into this branch, as it will remove the code